### PR TITLE
Fix #545, check REDISMODULE_CTX_FLAGS_DENY_BLOCKING flag before blocking the client.

### LINKED ISSUE #545: Redis crashes when switching from RG.TRIGGER to "keymiss" event for RedisAI model
Steps to reproduce:
1. Checkout https://github.com/applied-knowledge-systems/the-pattern-platform 
2. Launch [Launch Cluster Demo](https://github.com/applied-knowledge-systems/the-pattern-platform/blob/main/conf/launch_cluster_docker_demo.sh)
3. Create virtualenv & populate RedisGears cluster by running 'sh cluster_pipeline.sh'
4. Checkout API with BERT QA models: https://github.com/applied-knowledge-systems/the-pattern-api/tree/qa_bert_geared
5. Load models into each shard `cd /the-pattern-api/qasearch && python export_load_bert.py`
6. Pre-compute tensors `gears-cli run --host 127.0.0.1 --port 30001 tokeniser_gears_redisai.py --requirements requirements.txt`
7. Register trigger function `gears-cli run --host 127.0.0.1 --port 30001 qa_redisai_gear_map.py` 
8. Validate trigger function works:
```python
startup_nodes = [{"host": "127.0.0.1", "port": "30001"}, {"host": "127.0.0.1", "port":"30002"}, {"host":"127.0.0.1", "port":"30003"}]
from redisai import ClusterClient
redisai_cluster_client = ClusterClient(startup_nodes=startup_nodes)
question="Effectiveness of community contact reduction"
sentence_key="PMC261870.xml:{06S}:26"
slot = redisai_cluster_client.connection_pool.nodes.keyslot(sentence_key)
node = redisai_cluster_client.connection_pool.get_master_node_by_slot(slot)
connection = redisai_cluster_client.connection_pool.get_connection_by_node(node)
connection.send_command('RG.TRIGGER',"RunQABERT",sentence_key,question)
print(redisai_cluster_client.parse_response(connection,"RG.TRIGGER"))
```
shall return "effectiveness of community contact reduction", 
9. Register gears with keymiss event: 'gears-cli run --host 127.0.0.1 --port 30001 qa_redisai_gear_map_keymiss.py'
10. Run: redis-cli -c -p 30001 -h 127.0.0.1 get "cache{06S}{PMC261870.xml:{06S}:26}_Laser Correction" 

[Crash report](https://gist.github.com/AlexMikhalev/ec56d96658093e91942ee36e30774a55).

### PR DESCRIPTION
**Pull request by MeirShpilraien**
Fix #545, check REDISMODULE_CTX_FLAGS_DENY_BLOCKING flag before blocking the client.

We check it on 3 places and act as if we run in MULTI/EXEC or Lua script:
1. RG.PYEXECUTE
2. Command override
3. Keys Reader commands option

### CODE REVIEW

File: plugins/python/redisgears_python.c [hunk 1]
```diff
@@ -4604,8 +4604,8 @@ static void RedisGearsPy_BackgroundExecute(PythonSessionCtx* session,
 int RedisGearsPy_Execute(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
     int ctxFlags = RedisModule_GetContextFlags(ctx);
 
-    if(ctxFlags & (REDISMODULE_CTX_FLAGS_LUA|REDISMODULE_CTX_FLAGS_MULTI)){
-        RedisModule_ReplyWithError(ctx, "Can not run gear inside multi exec or lua");
+    if(ctxFlags & (REDISMODULE_CTX_FLAGS_LUA|REDISMODULE_CTX_FLAGS_MULTI|REDISMODULE_CTX_FLAGS_DENY_BLOCKING)){
+        RedisModule_ReplyWithError(ctx, "Can not run gear inside multi exec, lua, or if blocking is not allowed");
```
Comment (chayim) [inline plugins/python/redisgears_python.c:4608]:
```
Can not run gears inside a multi exec, lua, or when blocking is not allowed
```

File: src/readers/command_reader.c [hunk 2]
```diff
@@ -1008,9 +1008,10 @@ static int CommandReader_Trigger(RedisModuleCtx *ctx, RedisModuleString **argv,
     int runFlags = 0;
     int ctxFlags = RedisModule_GetContextFlags(ctx);
     if((ctxFlags & REDISMODULE_CTX_FLAGS_MULTI) ||
-                    (ctxFlags & REDISMODULE_CTX_FLAGS_LUA)){
+       (ctxFlags & REDISMODULE_CTX_FLAGS_LUA) ||
+       (ctxFlags & REDISMODULE_CTX_FLAGS_DENY_BLOCKING)){
         if(crtCtx->mode != ExecutionModeSync){
-            RedisModule_ReplyWithError(ctx, "ERR can not run a none sync execution inside MULTI/LUA or on loading.");
+            RedisModule_ReplyWithError(ctx, "ERR can not run a none sync execution inside MULTI/LUA (blocking is not allowed) or on loading.");
```
Comment (chayim) [inline src/readers/command_reader.c:1014]:
```
ERR can not run a non-sync execution inside a MULTI/LUA (blocking is not allowed) or on loading.
```

File: plugins/python/redisgears_python.c (hunk 1 above)
Comment (MeirShpilraien) [reply plugins/python/redisgears_python.c]: Thanks, fixed.

File: src/readers/command_reader.c (hunk 2 above)
Comment (MeirShpilraien) [reply src/readers/command_reader.c]: Thanks, fixed.

**Review by chayim — APPROVED** [review]

**Pull request merged**